### ISSUE TITLE
CMake fails to find XCTest (3.0.1 release)

### ISSUE THREAD

**Issue by AudioScientist** [leetal/ios-cmake#37]
Hello,

I tried the recent 3.0.1 update, and encountered an issue I can't fix.

For some part of my code, I use CMake to create XCTest targets with your toolchain using XCode generator. For this, I add `find_package(XCTest REQUIRED)` in my CMakeLists, and use the functions `xctest_add_bundle` and `xctest_add_test` to generator my XCTest targets.

Everything worked fine, but when I updated the toolchain from 2.2.0 to the 3.0.1 and changed `IOS_PLATFORM` to `PLATFORM` in my Cmake build scrits, I obtained the following error : 
```
CMake Error at /usr/local/Cellar/cmake/3.14.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find XCTest (missing: XCTest_LIBRARY XCTest_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.14.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.14.1/share/cmake/Modules/FindXCTest.cmake:84 (find_package_handle_standard_args)
  MWMAudioKit/MWMAudioKit_Core/CoreMathVector-Tests/CMakeLists.txt:30 (find_package)
```

I'm running CMake 3.14.1 and I have XCode 10.2 with iOS SDK 12.2 installed.

I looked at the diff of your release and tried a few things but couldn't find any solution.
Any idea of something I or you did wrong ?

Thanks again for your work :)

Tom

**Comment by leetal** [comment]
Hmmm.. I cannot remember changing anything that would break find_package. 
I'll look into this during the coming days and see what I can find. Obviously a test with find_package is needed as well. :)

**Comment by leetal** [comment]
So i think i have found it. Newer versions of CMake (3.14+) has first-class support for iOS (and watchOS + tvOS, ..) and thus requires `CMAKE_SYSTEM_NAME` to be set to the respective platform. I have modified the toolchain to adhere to that variable, but when doing so it seems to have broken the FindXCTest.cmake module which somehow requires `CMAKE_SYSTEM_NAME` to be set to `Darwin` (which it is not if not running older versions of CMake than 3.14). A temporary fix would probably be to set `CMAKE_SYSTEM_NAME` to `Darwin` (`set(CMAKE_SYSTEM_NAME Darwin FORCE)`) just before the call to `find_package(XCTest REQUIRED)` (untested though).

**Comment by leetal** [comment]
I will ask Kitware about this and see what they come up with.

**Comment by AudioScientist** [comment]
Thanks for finding some time to work on the issue. I didn't have time to investigate myself yet. I just tested the `set(CMAKE_SYSTEM_NAME Darwin FORCE)` on my targets using XCTest but it doesn't seem to work. I also tried to comment [lines 421 to 436](https://github.com/leetal/ios-cmake/blob/76134c1a97bd18d5c2e83d620c676fdd4b29ad52/ios.toolchain.cmake#L421) (and 439) but it didn't help either.

I'll stick with the previous version for now.

I'll try to find some time to investigate a bit and/or make a simple example to illustrate the issue.

**Comment by leetal** [comment]
Update: Kitware is aware of the issue and are working on a solution for newer versions of Cmake.

**Comment by leetal** [comment]
BTW, if this is still an issue: find_host_package(XCtest REQUIRED) will probably fix the issue in the meantime while Kitware fixes the main issue.

**Comment by leetal** [comment]
Kitware issue tracker here: https://gitlab.kitware.com/cmake/cmake/issues/19172#note_567996

**Comment by leetal** [comment]
Kitware have released 3.14.4 a few days ago that fixes this issue. Closing.

**Comment by AudioScientist** [comment]
Perfect, just did the upgrade, works like a charm, thanks for the update :)
